client/dvm/rpc/src/debug.rs

```diff
@@ -258,15 +250,19 @@ where
 				tracer: Some(tracer),
 				..
 			}) => {
-				let hash: H128 = sp_io::hashing::twox_128(&tracer.as_bytes()).into();
-				let blockscout_hash = H128::from_str("0x94d9f08796f91eb13a2e82a6066882f7").unwrap();
-				let tracer = if hash == blockscout_hash {
-					Some(TracerInput::Blockscout)
-				} else if tracer == "callTracer" {
-					Some(TracerInput::CallTracer)
-				} else {
-					None
-				};
+				const BLOCKSCOUT_JS_CODE_HASH: [u8; 16] =
+					hex_literal::hex!("94d9f08796f91eb13a2e82a6066882f7");
+				const BLOCKSCOUT_JS_CODE_HASH_V2: [u8; 16] =
+					hex_literal::hex!("89db13694675692951673a1e6e18ff02");
```

Comment:
What's these magic hashes used for?

Reply:
These hashes are used to support multiple callTracers.
 
For example, you can send debug request like
`{ "jsonrpc":"2.0", "id":1, "method":"debug_traceBlockByNumber", "params": ["0xbf", {"tracer": "callTracer"}]}` Or 
https://github.com/PureStake/moonbeam/blob/master/tests/tracing-tests/test-trace.ts#L196-L217.

I do not fully understand the reason they use JSON hash instead of String as param. I guess this is the requirement of [Blockscout](https://blockscout.com/).